### DISCUSSION TITLE
How convert rtsp stream to rtmp stream

### DISCUSSION THREAD

**Discussion by fanshun** [bluenviron/mediamtx#301] [Questions]
## Which version are you using?

v0.15.0

## Question
I push rtsp stream with ffmpeg，and then I would convert rtsp stream to rtmp stream internally, does it surpport?

ffmpeg  -re -stream_loop -1 -i ./video/bigbuck.ts -vcodec copy -codec copy -rtsp_transport tcp -f rtsp rtsp://localhost:8554/222

##Description
According the doc, my config is as follows:
runOnPublish: ffmpeg -f rtsp -rtsp_transport tcp -i rtsp://localhost:8554/222 -f flv -an rtmp://localhost:1935/live/111

The erros are as follows:
2021/03/17 15:23:12 [I] [2/1/0] [client 127.0.0.1:46438] connected (RTSP/TCP)
2021/03/17 15:23:12 [I] [2/1/1] [client 127.0.0.1:46438] is reading from path '222', 2 tracks with udp
[mpeg2video @ 0x55b4d3f65840] Invalid frame dimensions 0x0.
    Last message repeated 10 times
Input #0, rtsp, from 'rtsp://localhost:8554/222':
  Metadata:
    title           : Stream
  Duration: N/A, start: -0.008689, bitrate: N/A
    Stream #0:0: Video: mpeg2video (Main), yuv420p(tv, bottom first), 1280x720 [SAR 1:1 DAR 16:9], 30 fps, 30 tbr, 90k tbn, 60 tbc
    Stream #0:1: Audio: mp2, 48000 Hz, stereo, s16p, 384 kb/s
2021/03/17 15:23:13 [I] [3/1/1] [client 127.0.0.1:36948] connected (RTMP)
Stream mapping:
  Stream #0:0 -> #0:0 (mpeg2video (native) -> flv1 (flv))
Press [q] to stop, [?] for help
Output #0, flv, to 'rtmp://localhost:1935/live/111':
  Metadata:
    title           : Stream
    encoder         : Lavf57.83.100
    Stream #0:0: Video: flv1 (flv) ([2][0][0][0] / 0x0002), yuv420p, 1280x720 [SAR 1:1 DAR 16:9], q=2-31, 200 kb/s, 30 fps, 1k tbn, 30 tbc
    Metadata:
      encoder         : Lavc57.107.100 flv
    Side data:
      cpb: bitrate max/min/avg: 0/0/200000 buffer size: 0 vbv_delay: -1
2021/03/17 15:23:13 [I] [3/1/1] [client 127.0.0.1:36948] ERR: unsupported video codec 2
2021/03/17 15:23:13 [I] [2/1/1] [client 127.0.0.1:36948] disconnected
av_interleaved_write_frame(): Broken pipe
[flv @ 0x55b4d3fcd760] Failed to update header with correct duration.
[flv @ 0x55b4d3fcd760] Failed to update header with correct filesize.
Error writing trailer of rtmp://localhost:1935/live/111: Broken pipe

Best wishes!

**Comment by aler9** [comment]
Hello, conversion is automatic, there's no need to add ffmpeg...
You can publish a stream with RTSP and pull it with RTMP.

**Comment by fanshun** [comment]
Could you give an example about the configuration? Thanks a lot. @aler9

**Comment by aler9** [comment]
The configuration is the default one.

Just publish your video to a path of your choice, with RTSP, to the server

```
ffmpeg -re -stream_loop -1 -i ./video/bigbuck.ts -vcodec copy -codec copy -rtsp_transport tcp -f rtsp rtsp://localhost:8554/mypath
```

and pull it with RTMP from the server

```
ffmpeg -re -i rtmp://localhost:1935/mypath ...
```

[a comment by an automated account is not shown]
